Cython/Compiler/Options.py

```diff
@@ -651,7 +651,7 @@ def __init__(self, defaults=None, **kw):
             import warnings
             warnings.warn("C++ mode forced when in Pythran mode!")
             options['cplus'] = True
-        if 'language_level' in directives and 'language_level' not in kw:
+        if directives.get('language_level', None) and 'language_level' not in kw:
```

Comment:
We don't need the explicit `None` here. And checking for `in` is faster than calling `.get()`, so let's reverse the order.
```suggestion
        if 'language_level' not in kw and directives.get('language_level'):
```